### PR TITLE
Removes Runscanner

### PR DESCRIPTION
**Pull request by avarsava**
Removes Runscanner code from MISO and adds dependency on stand-alone
Runscanner.

Please see oicr-gsi/runscanner for new Runscanner code.

### CODE REVIEW

**Comment by djcooke** [comment]
Should not be merged until Runscanner has a release version that is deployed to Artifactory (at which point, tests will be able to run on Travis too).

Merging develop into this branch has not resolved the conflicts either. It may be easiest to create a new branch from develop and cherry-pick your commit onto it?

**Comment by avarsava** [comment]
See also: https://github.com/oicr-gsi/runscanner/pull/2

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java [hunk 1]
```diff
@@ -2870,4 +2871,43 @@ private static void setId(Consumer<Long> setter, Identifiable value) {
...
+  }
+
+  /**
+   * Converts ABCD to Abcd for compatibility with MISO's enum formatting.
+   * 
+   * @param name Value of a Runscanner enum, all caps
+   * @return name made Upper Camel Case
+   */
+  private static String toMisoFormat(String name) {
+    return name.substring(0, 1) + name.substring(1).toLowerCase();
```
Comment (apmasell) [inline miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java]: I really think this should be a switch block. It's much easier to understand

Comment (djcooke) [reply miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java]: Or just capitalize the enum values in MISO too so this is unneccessary

Comment (avarsava) [reply miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java]: @djcooke I tried that, it broke a lot of database things so I figured this was the best route for now

@apmasell Could you please elaborate? I currently don't see how there's any decision-making in this function

Comment (djcooke) [reply miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java]: You would have to include a DB migration that updates the DB values to match the enum updates

File: docs/_posts/2016-01-11-installation-guide.md [hunk 2]
```diff
@@ -300,60 +293,14 @@ to allow duplicate library aliases. A custom validator must be specified for thi
 property to be enabled - the naming scheme’s default validator will not be altered.
 
 # Setting Up the Run Scanner
-The run scanner is a webservice that scans the paths containing
+Run Scanner is a webservice that scans the paths containing
```
Comment (callunity) [inline docs/_posts/2016-01-11-installation-guide.md]: Can you please link this to the GitHub repo?

File: docs/_posts/2016-01-11-installation-guide.md [hunk 3]
```diff
@@ -300,60 +293,14 @@ to allow duplicate library aliases. A custom validator must be specified for thi
...
-      },
-      {
-        "path": "/srv/sequencer/hiseq2500_2",
-        "platformType": "ILLUMINA",
-        "name": "default",
-        "timeZone": "America/Toronto",
-        "parameters": {}
-      }
-    ]
+Once complete, edit `$CATALINA_HOME/conf/Catalina/localhost/miso.properties` of the MISO Tomcat server and set `miso.runscanner.urls` to the URL of the Run Scanner instance. It is possible to set up multiple run scanners managing different sequencers and add all the URLs to `miso.properties`. Restart MISO.
```
Comment (callunity) [inline docs/_posts/2016-01-11-installation-guide.md]: "Restart MISO" may case some confusion, as at this point in the guide users haven't yet learned how to start MISO.

File: docs/_posts/2016-01-11-installation-guide.md (hunk 2 above)
Comment (callunity) [reply docs/_posts/2016-01-11-installation-guide.md]: Bump

**Pull request merged**